library/modules/Military.cpp

```diff
@@ -0,0 +1,304 @@
+#include <string>
+#include <vector>
+#include <array>
+#include "modules/Military.h"
+#include "modules/Translation.h"
+#include "df/building.h"
+#include "df/building_civzonest.h"
+#include "df/historical_figure.h"
+#include "df/historical_entity.h"
+#include "df/entity_position.h"
+#include "df/entity_position_assignment.h"
+#include "df/plotinfost.h"
+#include "df/squad.h"
+#include "df/squad_position.h"
+#include "df/squad_schedule_order.h"
+#include "df/squad_order.h"
+#include "df/squad_order_trainst.h"
+#include "df/world.h"
+
+using namespace DFHack;
+using namespace df::enums;
+using df::global::world;
+using df::global::plotinfo;
+
+std::string Military::getSquadName(int32_t squad_id)
+{
+    df::squad *squad = df::squad::find(squad_id);
+    if (!squad)
+        return "";
+    if (squad->alias.size() > 0)
+        return squad->alias;
+    return Translation::TranslateName(&squad->name, true);
+}
+
+//only works for making squads for fort mode player controlled dwarf squads
+//could be extended straightforwardly by passing in entity
+df::squad* Military::makeSquad(int32_t assignment_id)
+{
+    if (df::global::squad_next_id == nullptr || df::global::plotinfo == nullptr)
+        return nullptr;
+
+    df::language_name name;
+    name.type = df::language_name_type::Squad;
+
+    for (int i=0; i < 7; i++)
+    {
+        name.words[i] = -1;
+        name.parts_of_speech[i] = df::part_of_speech::Noun;
+    }
+
+    df::historical_entity* fort = df::historical_entity::find(df::global::plotinfo->group_id);
+
+    df::entity_position_assignment* found_assignment = nullptr;
+
+    for (auto* assignment : fort->positions.assignments)
+    {
+        if (assignment->id == assignment_id)
+        {
+            found_assignment = assignment;
+            break;
+        }
+    }
+
+    if (found_assignment == nullptr)
+        return nullptr;
+
+    //this function does not attempt to delete or replace squads for assignments
+    if (found_assignment->squad_id != -1)
+        return nullptr;
+
+    df::entity_position* corresponding_position = nullptr;
+
+    for (auto* position : fort->positions.own)
+    {
+        if (position->id == found_assignment->position_id)
+        {
+            corresponding_position = position;
+            break;
+        }
+    }
+
+    if (corresponding_position == nullptr)
+        return nullptr;
+
+    df::squad* result = new df::squad();
+    result->id = *df::global::squad_next_id;
+    result->cur_routine_idx = 0;
+    result->uniform_priority = result->id + 1; //no idea why, but seems to hold
```

Comment:
yes, this is curious. what information is `uniform_priority` supposed to convey?